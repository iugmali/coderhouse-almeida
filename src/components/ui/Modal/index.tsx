'use client'

import {ReactNode} from "react";
import { motion } from "framer-motion";

type Props = {
  onClose: () => void
  children: ReactNode
}

const Modal = ({onClose, children}: Props) => {
  return (
    <>
        <motion.div
          key={'modal-backdrop'}
          initial={{
            opacity: 1
          }}
          animate={{
            opacity: 0.7
          }}
          transition={{duration: 1}}
          exit={{
            opacity: 0,
            transition: {duration: 0.3}
          }}
          className={`fixed top-0 z-[60] h-screen w-screen bg-gray-950`} onClick={() => onClose()}/>
        <motion.div
          key={'modal'}
          initial={{
            y: -30,
            opacity: 0
          }}
          animate={{
            y: 0,
            opacity: 1
          }}
          exit={{
            y: -30,
            opacity: 0,
            transition: {duration: 0.3}
          }}
          className={`z-[60] p-2 fixed top-1/4 left-2 right-2 max-w-md mx-auto bg-gray-50 grid place-content-center rounded`}
        >
          {children}
        </motion.div>
    </>
  )
}

export default Modal
