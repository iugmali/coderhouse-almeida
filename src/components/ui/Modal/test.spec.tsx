import {render} from "@testing-library/react";
import Modal from "@/components/ui/Modal/index";

describe('Modal', () => {
  it('renders unchanged', () => {
    const {container} = render(<Modal onClose={jest.fn()}>modal</Modal>)
    expect(container.firstChild).toMatchSnapshot();
  })
})
