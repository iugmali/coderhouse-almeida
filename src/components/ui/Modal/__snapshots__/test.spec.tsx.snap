// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders unchanged 1`] = `
<div
  class="fixed top-0 left-0 z-[60] h-screen w-screen bg-gray-950"
  style="opacity: 1;"
/>
`;
